nomad/job_endpoint_test.go

```diff
@@ -6651,6 +6651,61 @@ func TestJobEndpoint_ValidateJobUpdate_ACL(t *testing.T) {
 	require.Equal("", validResp.Warnings)
 }
 
+func TestJobEndpoint_ValidateJob_PriorityNotOk(t *testing.T) {
+	ci.Parallel(t)
+
+	s1, cleanupS1 := TestServer(t, nil)
+	defer cleanupS1()
+	codec := rpcClient(t, s1)
+	testutil.WaitForLeader(t, s1.RPC)
+
+	validateJob := func(j *structs.Job) error {
+		req := &structs.JobRegisterRequest{
+			Job: j,
+			WriteRequest: structs.WriteRequest{
+				Region:    "global",
+				Namespace: j.Namespace,
+			},
+		}
+		var resp structs.JobValidateResponse
+		if err := msgpackrpc.CallWithCodec(codec, "Job.Validate", req, &resp); err != nil {
+			return err
+		}
+
+		if resp.Error != "" {
+			return errors.New(resp.Error)
+		}
+
+		if len(resp.ValidationErrors) != 0 {
+			return errors.New(strings.Join(resp.ValidationErrors, ","))
+		}
+
+		if resp.Warnings != "" {
+			return errors.New(resp.Warnings)
+		}
+
+		return nil
+	}
+
+	t.Run("job with invalid min priority", func(t *testing.T) {
+		j := mock.Job()
+		j.Priority = 0
+
+		err := validateJob(j)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "job priority must be between")
+	})
+
+	t.Run("job with invalid max priority", func(t *testing.T) {
+		j := mock.Job()
+		j.Priority = 101
+
+		err := validateJob(j)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "job priority must be between")
```

Comment:
We're gradually migrating over to https://github.com/shoenig/test/ which has much better equality detection. So new tests should use that where possible:

```suggestion
		must.Error(t, err)
		must.ErrorContains(t, err, "job priority must be between")
```

(similarly for the other new tests you've introduced in this PR)